libr/anal/p/anal_mips_gnu.c

```diff
@@ -277,46 +1446,138 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b, int len, R
 		// TODO: this is a stop-gap. Really we need some smarts in here to tie this into the
 		// flags directly, as suggested here: https://github.com/radareorg/radare2/issues/949#issuecomment-43654922
 		case 15: // lui
+			insn.id = MIPS_INS_LUI;
+			sprintf (insn.i_reg.imm, "0x%"PFMT64x, imm);
 			op->dst = r_anal_value_new ();
-			op->dst->reg = r_reg_get (anal->reg, mips_reg_decode(rt), R_REG_TYPE_GPR);
+			op->dst->reg = r_reg_get (anal->reg, mips_reg_decode (rt), R_REG_TYPE_GPR);
 			// TODO: currently there is no way for the macro to get access to this register
 			op->val = imm;
 			break;
 		case 9: // addiu
+			insn.id = MIPS_INS_ADDIU;
+			op->type = R_ANAL_OP_TYPE_ADD;
 			op->dst = r_anal_value_new ();
 			op->dst->reg = r_reg_get (anal->reg, mips_reg_decode(rt), R_REG_TYPE_GPR);
 			// TODO: currently there is no way for the macro to get access to this register
 			op->src[0] = r_anal_value_new ();
 			op->src[0]->reg = r_reg_get (anal->reg, mips_reg_decode(rs), R_REG_TYPE_GPR);
 			op->val = imm; // Beware: this one is signed... use `?vi $v`
+			if (rs == 0) {
+				insn.id = MIPS_INS_LI;
+				sprintf (insn.i_reg.imm, "0x%"PFMT32x, imm);
+			}
 			break;
 		case 8: // addi
-		case 10: // stli
-		case 11: // stliu
+			insn.id = MIPS_INS_ADDI;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			break;
+		case 10: // slti
+			insn.id = MIPS_INS_SLTI;
+			break;
+		case 11: // sltiu
+			insn.id = MIPS_INS_SLTIU;
+			break;
 		case 12: // andi
+			insn.id = MIPS_INS_ANDI;
+			op->type = R_ANAL_OP_TYPE_AND;
+			break;
 		case 13: // ori
+			insn.id = MIPS_INS_ORI;
+			op->type = R_ANAL_OP_TYPE_OR;
+			break;
 		case 14: // xori
+			insn.id = MIPS_INS_XORI;
+			op->type = R_ANAL_OP_TYPE_XOR;
+			break;
+		case 24: // daddi
+			insn.id = MIPS_INS_DADDI;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			break;
+		case 25: // daddiu
+			insn.id = MIPS_INS_DADDIU;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			if (rs == 0) {
+				insn.id = MIPS_INS_LDI;
+				sprintf (insn.i_reg.imm, "0x%"PFMT64x, imm);
+			}
+			break;
 		case 32: // lb
+			op->refptr =  1;
+			insn.id = MIPS_INS_LB;
 		case 33: // lh
+			if (!op->refptr) {
+				op->refptr =  2;
+				insn.id = MIPS_INS_LB;
+			}
 		case 35: // lw
+			if (!op->refptr) {
+				op->refptr =  4;
+				insn.id = MIPS_INS_LW;
+			}
```

Comment:
if this fallback is on purpose, use /* fallthrough */ comment instead of "break" to avoid static-analysis tools to complain